src/Entity/Form/SubscriptionForm.php

```diff
@@ -126,7 +156,116 @@ public function save(array $form, FormStateInterface $form_state) {
     }
     catch (\Exception $e) {
       $this->messenger->addError($e->getMessage());
+      $messages = $this->messenger->all();
+      $this->messenger->deleteAll();
+      if (!empty($messages['error'][0])) {
+        $overlap_error = $messages['error'][0];
+        // Make sure this is overlapping error message.
+        if (strstr($overlap_error, 'has following overlapping')) {
```

Comment:
Shouldn't we be checking the client error with something like:

```
if (($client_error = $e->getPrevious())
  && $client_error instanceof ClientErrorException
  && $client_error->getEdgeErrorCode() === 'mint.developerHasFollowingOverlapRatePlans'
) {
```

Reply:
I guess Minnur found this from the legacy code. This is how they implemented that.

This is much better. Nice find. I'll implement this.